src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1861,4 +1866,53 @@ void InterpreterCreateQuery::addColumnsDescriptionToCreateQueryIfNecessary(ASTCr
     }
 }
 
+void InterpreterCreateQuery::processSQLSecurityOption(ContextPtr context_, ASTSQLSecurity & sql_security, bool is_attach)
+{
+    if (!sql_security.type.has_value())
+    {
+        /// SQL SECURITY/DEFINER options wasn't specified, will use defaults.
+        String default_security = context_->getSettingsRef().default_view_sql_security;
+
+        if (default_security == "NONE")
+            sql_security.type = ASTSQLSecurity::Type::NONE;
+        else if (default_security == "INVOKER")
+            sql_security.type = ASTSQLSecurity::Type::INVOKER;
+        else if (default_security == "DEFINER")
+        {
+            sql_security.type = ASTSQLSecurity::Type::DEFINER;
+            String default_definer = context_->getSettingsRef().default_view_definer;
+            if (default_definer == "CURRENT_USER")
+                sql_security.is_definer_current_user = true;
+            else
+                sql_security.definer = std::make_shared<ASTUserNameWithHost>(default_definer);
+        }
+        else
+            throw Exception(ErrorCodes::SYNTAX_ERROR, "Expected default_view_sql_security = <DEFINER | INVOKER | NONE>. Got {}", default_security);
+    }
+
+    const auto current_user_name = context_->getUserName();
+    if (sql_security.is_definer_current_user)
+    {
+        if (current_user_name.empty())
+            /// This can happen only when attaching a view for the first time after migration and with `CURRENT_USER` default.
+            sql_security.type = ASTSQLSecurity::Type::NONE;
```

Comment:
Shouldn't we set here `INVOKER` for ordinary views and `NONE` for materialized views to be consistent with older behaviour?